soc/realtek/ec/rts5912/reg/reg_uart.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2023 Realtek, SIBG-SD7
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _REALTEK_RTS5912_REG_UART_H
+#define _REALTEK_RTS5912_REG_UART_H
+
+/*
+ * @brief UART Controller (UART)
+ */
+
+typedef struct {
```

Comment:
this looks like ns16550 UART, there's an existing driver for it. am I wrong?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/75267#discussion_r1775955793 
We have discussed this item before. We will survey it and it will be our future work, thanks.

Reply:
@gmarull, in my opinion extending the ns16550 UART driver to support this SoC is better done as a separate PR so that the ns16550 maintainers can properly review the changes.

Reply:
> @gmarull, in my opinion extending the ns16550 UART driver to support this SoC is better done as a separate PR so that the ns16550 maintainers can properly review the changes.

Disregard, I see the author already refactored the commit to use the ns16550